lms/static/scripts/frontend_apps/errors.js

```diff
@@ -8,8 +8,26 @@ export class PickerCanceledError extends Error {
 }
 
 /**
- * Error returned when an API call fails with a 4xx or 5xx response and
- * JSON body.
+ * @typedef {import('./config').ConfigErrorBase} ConfigErrorBase
+ */
+
+/**
+ * Error returned when error data is provided in application configuration JSON
+ * and application is in an error mode ('error-dialog' or 'oauth2-redirect-error')
+ */
+export class AppConfigError extends Error {
```

Comment:
To me this class name makes it seem there is a problem with the app's configuration rather than a backend error which the frontend is being asked to render. This error seems conceptually similar to `APIError` except that the error occurred in a backend operation that occurred before the frontend was launched, rather than in an error processing a request initiated by the frontend.

Maybe something like `BackendError` or `ServerError` would make sense?

Reply:
Yeah, I struggled with this naming, too. Your ideas sound sensible.